### PR TITLE
Added lazy loading of static asset information

### PR DESCRIPTION
**Pull request by noisecapella**

### CODE REVIEW

**Comment by carsongee** [comment]
Just happened to come by, looks like the package is missing from bower

**Comment by carsongee** [comment]
and now I know why from #526

**Comment by pdpinch** [comment]
This is working well for me.

File: ui/static/ui/js/utils.jsx [hunk 1]
```diff
@@ -1,12 +1,75 @@
...
+      });
+    },
+    handleInfiniteLoad: function () {
+      this.setState({
+        isInfiniteLoading: true
+      });
+      this.loadPage();
+    },
+    render: function () {
+      var spinner = <div className="infinite-item-list">Loading...</div>;
```
Comment (giocalitri) [inline ui/static/ui/js/utils.jsx:50]: This can be leaved as it is and maybe it's out of scope, but isn't a real spinner better?

Comment (noisecapella) [inline ui/static/ui/js/utils.jsx:50]: This was pulled from their example code, I didn't do much thinking on how this would look. I have no argument switching to a spinner it may be better to do it in a separate PR

File: ui/static/ui/js/static_assets.jsx [hunk 2]
```diff
@@ -1,8 +1,15 @@
...
+  ], function (React, $, _, Utils) {
   'use strict';
 
   var StatusBox = Utils.StatusBox;
+  var InfiniteList = Utils.InfiniteList;
+
+  // Guess at a good value for the height of each element. Infinite requires
+  // an elementHeight for each item. If we wanted to do a per item elementHeight
+  // we could make this a list instead.
+  var elementHeight = 50;
```
Comment (giocalitri) [inline ui/static/ui/js/static_assets.jsx:12]: how did you determine this number is appropriate? trying to understand the logic

File: ui/static/ui/js/static_assets.jsx [hunk 3]
```diff
@@ -10,49 +17,49 @@ define('static_assets', [
...
-            <a href={asset.asset}>{asset.name}</a>
-          </li>;
-      });
+      var url = "/api/v1/repositories/" + this.props.repoSlug +
+          "/learning_resources/" +
+          this.props.learningResourceId + "/static_assets/";
+
+      // Infinite requires you to specify a containerHeight. This value
+      // is a guess at how big the area should be relative to the height.
+      var containerHeight = $(window).height() - 200;
```
Comment (giocalitri) [inline ui/static/ui/js/static_assets.jsx:33]: same as before for the number (the 200)

File: ui/static/ui/js/static_assets.jsx (hunk 2 above)
Comment (noisecapella) [inline ui/static/ui/js/static_assets.jsx:12]: It's just a guess, I picked a value that seemed to look good in the UI

File: ui/static/ui/js/static_assets.jsx (hunk 3 above)
Comment (noisecapella) [inline ui/static/ui/js/static_assets.jsx:33]: I was trying to pick a height which is the window height minus 200 pixels for the panel borders. It's also just a guess

File: ui/static/ui/js/static_assets.jsx (hunk 2 above)
Comment (giocalitri) [inline ui/static/ui/js/static_assets.jsx:12]: I'm fine with this, but I'm wondering if we need to test on all the browsers if it has the same behavior (same for the other number below)

Comment (giocalitri) [inline ui/static/ui/js/static_assets.jsx:12]: cc @pdpinch

**Comment by giocalitri** [comment]
The code looks fine to me other than two small comments: I'm going to test functionality

**Comment by giocalitri** [comment]
If you make any change to the code, please rebase master.

File: ui/static/ui/js/static_assets.jsx (hunk 2 above)
Comment (pdpinch) [inline ui/static/ui/js/static_assets.jsx:12]: I'll do some quick cross-browser testing on my Mac. 

I also saw it do some odd, but ultimately OK things, when my viewport was short (I had the developer panel open to look at the network traffic). 

> I'm wondering if we need to test on all the browsers if it has the same behavior (same for the other number below)

Comment (pdpinch) [inline ui/static/ui/js/static_assets.jsx:12]: Everything I've tested behaves the same. 

I will probably run this UI by the designer. I think there's probably some polish we can add, but it totally meets the needs of the story.

**Comment by giocalitri** [comment]
Functionality is fine: feel free to merge

**Comment by giocalitri** [comment]
(sorry it took so long, I wanted to re-run the tests locally)
